customer-managed/azure/terraform/resource_groups.tf

```diff
@@ -16,10 +16,15 @@ locals {
 }
 
 resource "azurerm_resource_group" "all" {
-  for_each = toset(local.resource_group_names)
+  for_each = var.create_resource_groups ? toset(local.resource_group_names) : []
 
   name     = each.value
   location = var.region
 
   tags = var.tags
 }
+
+data "azurerm_resource_group" "all" {
+  for_each = var.create_resource_groups ? [] : toset(local.resource_group_names)
```

Comment:
Is it possible that one of the names does not exist?

Reply:
If user chooses to provide RGs by setting `create_resource_groups` to `false`, the RGS must exist. If does not exist, the data source will fail.